core/src/main/java/io/aiven/kafka/tieredstorage/fetch/index/MemorySegmentIndexesCache.java

```diff
@@ -68,7 +70,9 @@ private static Weigher<SegmentIndexKey, byte[]> weigher() {
 
     protected AsyncCache<SegmentIndexKey, byte[]> buildCache(final CacheConfig config) {
         this.executor = config.threadPoolSize().map(ForkJoinPool::new).orElse(new ForkJoinPool());
+        new ThreadPoolMonitor(THREAD_POOL_METRIC_GROUP, executor);
```

Comment:
Does this need to be closed at some point? Or is everything handled by the GC?

Reply:
Yeah, no resources opened to close really. But should be _cleaned_ when closing the cache.